graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH6.kt

```diff
@@ -0,0 +1,142 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.HasCycle, GraphBehaviours.CreateDirectedGraph, """
+    Determine if one or more cycles exist in graph
+""")
+class BEH6 : GraphScenario() {
+
+    @Eac("When a directed graph has one cycle, return true")
+    fun hasOneCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+            }
+        }
+        then {
+            hasCycle(graphA, true)
```

Comment:
⛏️ Seeing it in use, I'm not sure this is the best name for this function. In terms of being as close to natural language as possible, I would have called it something like `containsCycles`.

I would at least use the plural form, `hasCycles`, because `hasCycle` makes me thing that the intention is to determine whether the graph contains a specific cycle, rather than any cycles.